### PR TITLE
[ENH]: Move func _mix_colormaps

### LINKED ISSUE #3499: Move _mix_colormaps() util function
This is a follow-up issue from #3173.

`plotting.html_surface._mix_colormaps()` should be moved to `plotting._utils` or `plotting.cm` in order to be more easily accessed from the rest of the codebase.

### PR DESCRIPTION
**Pull request by achamma723**
Closes #3499

### CODE REVIEW

**Comment by Remi-Gau** [comment]
hey @achamma723 thanks for the pull request

given the number of file / line changed, I suspect that you may have run black on some files that you should not have been formatted.

do you think you could just have a commit with just the function moved?

if you want to then apply black "safely", you can either:

- run black from the root of the repository

```
black .
```

by doing so black will read from its configuration in the `pyproject.toml` what file it should not act on

- let pre-commit handle all formatting

https://nilearn.github.io/stable/development.html#pre-commit

**Comment by achamma723** [comment]
> hey @achamma723 thanks for the pull request
> 
> given the number of file / line changed, I suspect that you may have run black on some files that you should not have been formatted.
> 
> do you think you could just have a commit with just the function moved?
> 
> if you want to then apply black "safely", you can either:
> 
> * run black from the root of the repository
> 
> ```
> black .
> ```
> 
> by doing so black will read from its configuration in the `pyproject.toml` what file it should not act on
> 
> * let pre-commit handle all formatting
> 
> https://nilearn.github.io/stable/development.html#pre-commit

Hello @Remi-Gau, in fact this was because I was using the black formatter (extension in vscode) which apparently is different than using the black function directly on the directory.
I'll manage to have a clean commit

**Comment by Remi-Gau** [comment]
ha yeah I had to disable the format on save feature of VScode for cases like this

sorry for the inconvenience

hopefully we should soon (ish ?) be fully black compatible

File: doc/changes/latest.rst [hunk 1]
```diff
@@ -60,3 +60,4 @@ Changes
 - Refactor error raising tests using context managers (:gh:`3854` BY `François Paugam`_)
 - Added warning to deprecate `darkness` in ``surf_plotting._compute_facecolors_matplotlib`` and ``html_surface._get_vertexcolor`` (:gh`3855` by `Alisha Kodibagkar`_)
 - :bdg-secondary:`Doc` Replace skipped doctests with default code-blocks (:gh:`3681` in by `Patrick Sadil`_)
+- Move the :func:`~nilearn.plotting.html_surface._mix_colormaps` in :mod:`~nilearn.plotting.cm` (:gh:`3919` by `Ahmad Chamma`_)
```
Comment (Remi-Gau) [inline doc/changes/latest.rst:63]: The build of the doc is getting the following warnings:

```
/home/runner/work/nilearn/nilearn/doc/changes/latest.rst:63: WARNING: py:func reference target not found: nilearn.plotting.html_surface._mix_colormaps
/home/runner/work/nilearn/nilearn/doc/changes/latest.rst:63: WARNING: py:mod reference target not found: nilearn.plotting.cm
```

The first one makes sense as the function does not exist anymore, so maybe remove the `:func:`.
Not sure about the second warning.

Comment (ymzayek) [reply doc/changes/latest.rst:63]: `_mix_colormaps` is private and not exposed so you can't link to it. Just do something like `` `_mix_colormaps` ``. To be explicit just mention the filenames you moved it from/to. `mod` role will work on the level of the modules like `plotting`, `maskers` etc. but not to link the python scripts inside those modules. Just use `` `cm.py` `` and be explicit that it's in the plotting module.

**Review by ymzayek — APPROVED** [review]
LGTM thanks!

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**